packages/orm/src/EntityManager.ts

```diff
@@ -139,24 +147,15 @@ export type LoaderCache = Record<string, DataLoader<any, any>>;
 type FilterAndOrder<T> = [FilterOf<T>, OrderOf<T> | undefined];
 
 export class EntityManager {
-  constructor(public knex: Knex) {}
+  constructor(public knex: Knex) {
+    this.contexty.create();
+  }
 
   private entities: Entity[] = [];
   private findLoaders: LoaderCache = {};
-  // TODO Extract this DataLoader + currentFlushPromise into its own abstraction
-  private flushLoader = new DataLoader<number, number>(
-    async (keys) => {
-      // Have all callers in the current event loop go through a single flush
-      // together, but set currentFlushPromise to let callers in subsequent loops
-      // that they need to wait.
-      this.currentFlushPromise = this.doFlush();
-      await this.currentFlushPromise;
-      this.currentFlushPromise = undefined;
-      return keys;
-    },
-    { cache: false },
-  );
-  private currentFlushPromise?: Promise<void>;
+  private flushSecret: number = 0;
+  private _isFlushing: boolean = false;
+  private contexty = new Contexty();
```

Comment:
Er, this is the that that is a module-level const. And looking at the source, they only install the hook once, which is good, but I think this `new Contexty` is the equivalent of `private static final ThreadLocal foo = new ThreadLocal()` that only needs instantiated once and then within handles multiplexing whoever called it back their right result.

Reply:
Actually on second thought, I don't think this is accurate.   A module level contexty would not be able to hand back the right result in the case of multiple entity managers.